packages/@n8n/task-runner/src/task-runner.ts

```diff
@@ -167,16 +170,20 @@ export abstract class TaskRunner extends EventEmitter {
 			(Object.values(this.openOffers).length + Object.values(this.runningTasks).length);
 
 		for (let i = 0; i < offersToSend; i++) {
+			// Add a bit of randomness so that not all offers expire at the same time
+			const validForInMs = OFFER_VALID_TIME_MS + randomInt(500);
+			// Add a little extra time to account for latency
+			const validUntil = process.hrtime.bigint() + msToNs(validForInMs + OFFER_VALID_EXTRA_MS);
```

Comment:
Why do we use ms for the offer that we send but ns for the offer that we track locally?

Reply:
It's because `process.hrtime` [exposes](https://nodejs.org/api/process.html#processhrtimebigint) the high resolution time only as bigint. I think `number` doesn't have enough capacity to represent time in nanoseconds